### PR TITLE
Gh-1 authorization service

### PR DESCRIPTION
**Pull request by buhhu8**
Autorization was done with md5 hash.

### CODE REVIEW

File: LKUK1.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Shagbark) [inline LKUK1.iml:1]: You shouldn't add this file to your git repositories. It's the special file for IntellijIdea and so it must not be in git

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <version>1.0-SNAPSHOT</version>
+
+    <!-- https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-web -->
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.4.0</version>
+    </parent>
+
+<dependencies>
```
Comment (Shagbark) [inline pom.xml]: Please, clear your code. You can use Ctrl+Alt+L in Idea to format code

File: src/main/java/org/lk/controller/GreetingRequest.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package org.lk.controller;
+
+public class GreetingRequest {
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/GreetingRequest.java]: Can we name it like `AuthorizaitonRequest`?

File: src/main/java/org/lk/controller/GreetingController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+@RestController // @Controller
+@RequestMapping("/greeting")
+public class GreetingController {
+
+    @PostMapping("/login")
+    public Greeting authentication(@RequestBody GreetingRequest request) {
+        Greeting obj = new Greeting();
+        Boolean flag = false;
+        Map<String, String> loginAndPassword = new HashMap<>();
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/GreetingController.java]: Rewrite this variable as the class field and fill this map in a code block or you can create `public void` method and annotate it with `@PostContruct`

File: src/main/java/org/lk/controller/GreetingController.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        loginAndPassword.put("buhhu8", "d8578edf8458ce06fbc5bb76a58c5ca4");  //qwerty
+        loginAndPassword.put("dmitry","912ec803b2ce49e4a541068d495ab570");  //asdf
+        loginAndPassword.put("anton","fd2cc6c54239c40495a0d3a93b6380eb");  //zxcv
+        for(Map.Entry<String, String> entry: loginAndPassword.entrySet()) {
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/GreetingController.java]: Can you replace this loop with streamapi?

File: src/main/java/org/lk/controller/Greeting.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package org.lk.controller;
+
+public class Greeting {
+
+    private String pairs;
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/Greeting.java:5]: Please, clarify what does this field mean?

File: src/main/java/org/lk/controller/GreetingController.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+            }
+            else if (flag == false){
+                obj.setPairs("User not found");
+            }
+
+        }
+       return obj;
+    }
+
+    private static String md5ApacheExample(String text){
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/GreetingController.java]: Why are you make it `static`?

File: src/main/java/org/lk/controller/GreetingController.java [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RestController // @Controller
+@RequestMapping("/greeting")
```
Comment (Shagbark) [inline src/main/java/org/lk/controller/GreetingController.java]: I think that we can remove this part of URL.

**Review by Shagbark — CHANGES_REQUESTED** [review]

File: src/main/java/org/lk/controller/GreetingRequest.java (hunk 3 above)
Comment (buhhu8) [reply src/main/java/org/lk/controller/GreetingRequest.java]: Yes, we can use name like AuthorizaitonRequest

File: src/main/java/org/lk/controller/GreetingController.java (hunk 7 above)
Comment (buhhu8) [reply src/main/java/org/lk/controller/GreetingController.java]: Because I don't want, that somebody can override this method.

**Pull request merged**